apps/client/app/routes/login.tsx

```diff
@@ -0,0 +1,61 @@
+import { ActionFunctionArgs, LoaderFunctionArgs, json, redirect } from "@remix-run/node";
+import AuthForm from "~/components/AuthForm";
+import { isUserLoggedIn, login } from "~/services/auth.server";
+import { createUserSession } from "~/services/session.server";
+import { safeRedirect } from "~/utils/helpers";
+
+export default function Login() {
+  return (
+    <div className="flex flex-col mx-56 p-16">
+      <AuthForm type="login" />
+    </div>
+  );
+}
+
+export async function loader({ request }: LoaderFunctionArgs) {
+  if (await isUserLoggedIn(request)) {
+    return redirect("/emails");
+  }
+
+  return json({});
+}
+
+export async function action({ request }: ActionFunctionArgs) {
+  const formData = await request.formData();
+  const email = formData.get("email");
+  const password = formData.get("password");
+  const remember = formData.get("remember");
+  const redirectTo = safeRedirect(formData.get("redirectTo"));
+
+  if (typeof email !== "string" || email.length === 0) {
```

Comment:
I suggest to create additional function for `validateAuthentication` for separate better  responsabilities, with also a validation of email format also in server:

```typescript
const validateAuthentication = (email, password) => {
  const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;

  if (typeof email !== "string" || email.length === 0) {
    return { error: "Email is required." };
  }

  if (!emailPattern.test(email)) {
    return { error: "Invalid email format." };
  }

  if (typeof password !== "string" || password.length < 6) {
    return { error: "Password is required and must be at least 6 characters." };
  }

  return null;
};

```